icu4j/main/classes/core/src/com/ibm/icu/number/NumberRangeFormatterImpl.java

```diff
@@ -433,11 +435,48 @@ Modifier resolveModifierPlurals(Modifier first, Modifier second) {
         StandardPlural resultPlural = fPluralRanges.resolve(firstParameters.plural, secondParameters.plural);
 
         // Get and return the new Modifier
-        assert firstParameters.obj == secondParameters.obj;
-        assert firstParameters.signum == secondParameters.signum;
         Modifier mod = firstParameters.obj.getModifier(firstParameters.signum, resultPlural);
         assert mod != null;
         return mod;
     }
 
+    /**
+     * Returns whether this Modifier is *semantically equivalent* to the other Modifier;
+     * in many cases, this is the same as equal, but parameters should be ignored.
+     *
+     * In ICU4C, this is a function on the Modifier class.
```

Comment:
Why is it not a function on the Modifier class in Java as well?

Reply:
We are requiring Java 8 since ICU 72 :-)
If this is awkward on the interface, then maybe turn that into a base class?
Or leave it as is?

Just noticed that it's different from C++, and therefore requires changing more call sites.

Reply:
ping

Reply:
ok I changed it to Java 8 `default`